storage/test/transfer.test.js

```diff
@@ -0,0 +1,639 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+var proxyquire = require('proxyquire').noCallThru();
+var bucketName = 'foo';
+var jobName = 'transferJobs/123456789012345678';
+var transferOperationName = 'transferOperations/123456789012345678';
+
+function getSample () {
+  var transferJobMock = {};
+  var transferOperationMock = {};
+  var storagetransferMock = {
+    transferJobs: {
+      create: sinon.stub().callsArgWith(1, null, transferJobMock),
+      get: sinon.stub().callsArgWith(1, null, transferJobMock),
+      patch: sinon.stub().callsArgWith(1, null, transferJobMock),
+      list: sinon.stub().callsArgWith(1, null, { transferJobs: [transferJobMock] })
+    },
+    transferOperations: {
+      get: sinon.stub().callsArgWith(1, null, transferOperationMock),
+      pause: sinon.stub().callsArgWith(1, null, transferOperationMock),
+      resume: sinon.stub().callsArgWith(1, null, transferOperationMock),
+      list: sinon.stub().callsArgWith(1, null, { operations: [transferOperationMock] })
+    }
+  };
+  var googleapisMock = {
+    storagetransfer: sinon.stub().returns(storagetransferMock),
+    auth: {
+      getApplicationDefault: sinon.stub().callsArgWith(0, null, {})
+    }
+  };
+  return {
+    program: proxyquire('../transfer', {
+      googleapis: googleapisMock
+    }),
+    mocks: {
+      googleapis: googleapisMock,
+      storagetransfer: storagetransferMock,
+      transferJob: transferJobMock,
+      transferOperation: transferOperationMock
+    }
+  };
+}
+
+describe('storage:transfer', function () {
+  describe('jobs', function () {
+    describe('create', function () {
+      it('should create a transfer job', function () {
+        var description = 'description';
+        var sample = getSample();
+        var callback = sinon.stub();
+
+        sample.program.createTransferJob(bucketName, bucketName, '2016/08/11', '15:30', null, callback);
+
+        assert(callback.calledOnce, 'callback called once');
+        assert.equal(callback.firstCall.args.length, 2, 'callback received 2 arguments');
+        assert.ifError(callback.firstCall.args[0], 'callback did not receive error');
+        assert.strictEqual(callback.firstCall.args[1], sample.mocks.transferJob, 'callback received transfer job');
+
+        sample.program.createTransferJob(bucketName, bucketName, '2016/08/11', '15:30', description, callback);
+
+        assert(callback.calledTwice, 'callback called twice');
+        assert.equal(callback.secondCall.args.length, 2, 'callback received 2 arguments');
+        assert.ifError(callback.secondCall.args[0], 'callback did not receive error');
+        assert.strictEqual(callback.secondCall.args[1], sample.mocks.transferJob, 'callback received transfer job');
+        assert.equal(sample.mocks.storagetransfer.transferJobs.create.secondCall.args[0].resource.description, description, 'description was set');
```

Comment:
This seems to test description-setting as well as transfer job creation - perhaps split each of these into its own test?